fast64_internal/oot/oot_actor.py

```diff
@@ -86,116 +449,134 @@ def drawActorHeaderProperty(layout, headerProp, propUser, altProp, objName):
+	bl_idname = "object.oot_search_actor_id_enum_operator"
+	bl_label = "Select Actor"
+	bl_property = "actorID"
+	bl_options = {'REGISTER', 'UNDO'}
+
+	actorID : bpy.props.EnumProperty(items = ootEnumActorID, default = "ACTOR_PLAYER")
+	objName : bpy.props.StringProperty()
+
+	def execute(self, context):
+		obj = bpy.data.objects[self.objName]
+		detailedProp = obj.fast64.oot.actor
+
+		obj.ootActorProperty.actorID = self.actorID
+		detailedProp.actorID = self.actorID
+		for actorNode in root:
+			if actorNode.get('ID') == self.actorID:
+				detailedProp.actorKey = actorNode.get('Key')
+
+		bpy.context.region.tag_redraw()
+		self.report({'INFO'}, "Selected: " + self.actorID)
+		return {'FINISHED'}
+
+	def invoke(self, context, event):
+		context.window_manager.invoke_search_popup(self)
+		return {'RUNNING_MODAL'}
+
+class OOTActorPropertiesLegacy(bpy.types.PropertyGroup):
+	# Normal Actors
+	# We can't delete this (for now) as it'd ignore data in older blend files
 	actorID : bpy.props.EnumProperty(name = 'Actor', items = ootEnumActorID, default = 'ACTOR_PLAYER')
 	actorIDCustom : bpy.props.StringProperty(name = 'Actor ID', default = 'ACTOR_PLAYER')
 	actorParam : bpy.props.StringProperty(name = 'Actor Parameter', default = '0x0000')
 	rotOverride : bpy.props.BoolProperty(name = 'Override Rotation', default = False)
-	rotOverrideX : bpy.props.StringProperty(name = 'Rot X', default = '0')
-	rotOverrideY : bpy.props.StringProperty(name = 'Rot Y', default = '0')
-	rotOverrideZ : bpy.props.StringProperty(name = 'Rot Z', default = '0')
+	rotOverrideX : bpy.props.StringProperty(name = 'Rot X', default = '0x0')
+	rotOverrideY : bpy.props.StringProperty(name = 'Rot Y', default = '0x0')
+	rotOverrideZ : bpy.props.StringProperty(name = 'Rot Z', default = '0x0')
 	headerSettings : bpy.props.PointerProperty(type = OOTActorHeaderProperty)
 
-def drawActorProperty(layout, actorProp, altRoomProp, objName):
-	#prop_split(layout, actorProp, 'actorID', 'Actor')
+def drawActorProperty(layout, actorProp, altRoomProp, objName, detailedProp):
 	actorIDBox = layout.column()
-	#actorIDBox.box().label(text = "Settings")
-	searchOp = actorIDBox.operator(OOT_SearchActorIDEnumOperator.bl_idname, icon = 'VIEWZOOM')
-	searchOp.actorUser = "Actor"
-	searchOp.objName = objName
 
-	split = actorIDBox.split(factor = 0.5)
-	split.label(text = "Actor ID")
-	split.label(text = getEnumName(ootEnumActorID, actorProp.actorID))
+	if detailedProp.isActorSynced:
+		drawDetailedProperties('Actor Property', actorProp, actorIDBox, objName, \
+			OOT_SearchActorIDEnumOperator, 'actorID', 'actorParam', detailedProp, detailedProp.actorKey)
+		drawActorHeaderProperty(actorIDBox, actorProp.headerSettings, "Actor", altRoomProp, objName)
+	else:
+		sceneObj = getSceneObj(bpy.context.object)
+		if sceneObj is None: sceneName = 'Unknown'
+		else: sceneName = sceneObj.name
+		actorIDBox.box().label(text=f"Scene: '{sceneName}' Actors are not synchronised!")
```

Comment:
I think "actors data is from an old fast64 version" would be more understandable to the end user. (especially with the "upgrade now" button below, and ideally both of these shouldn't show up at all if the blender handler works)
(also I don't like "synchronised" itself personally)

Reply:
is ``Legacy data from scene: '{sceneName}' has not been upgraded!`` better?